src/Sarif.Viewer.VisualStudio/ErrorList/SarifSnapshot.cs

```diff
@@ -188,26 +163,31 @@ private void ErrorListInlineLink_Click(object sender, RoutedEventArgs e)
 
             if (hyperLink != null)
             {
-                Tuple<int, int> indices = hyperLink.Tag as Tuple<int, int>;
+                Tuple<object, int> data = hyperLink.Tag as Tuple<object, int>;
 
-                SarifErrorListItem sarifResult = _errors[indices.Item1];
+                SarifErrorListItem sarifResult = _errors[Convert.ToInt32(data.Item1)];
 
-                // Set the current sarif error in the manager so we track code locations.
-                CodeAnalysisResultManager.Instance.CurrentSarifError = sarifResult;
+                CodeFlowLocationModel location = sarifResult.RelatedLocations.Where(l => l.Id == data.Item2).FirstOrDefault();
 
-                if (sarifResult.HasDetails)
-                {
-                    // Setting the DataContext to be null first forces the TabControl to select the appropriate tab.
-                    SarifViewerPackage.SarifToolWindow.Control.DataContext = null;
-                    SarifViewerPackage.SarifToolWindow.Control.DataContext = sarifResult;
-                }
-                else
+                if (location != null)
                 {
-                    SarifViewerPackage.SarifToolWindow.Control.DataContext = null;
-                }
+                    // Set the current sarif error in the manager so we track code locations.
+                    CodeAnalysisResultManager.Instance.CurrentSarifError = sarifResult;
+
+                    if (sarifResult.HasDetails)
+                    {
+                        // Setting the DataContext to be null first forces the TabControl to select the appropriate tab.
+                        SarifViewerPackage.SarifToolWindow.Control.DataContext = null;
+                        SarifViewerPackage.SarifToolWindow.Control.DataContext = sarifResult;
+                    }
+                    else
+                    {
+                        SarifViewerPackage.SarifToolWindow.Control.DataContext = null;
```

Comment:
Since you do this in both branches, hoist it out of the conditional; then you don't need the `else` branch.

Reply:
Done